src/targets/github.ts

```diff
@@ -395,32 +416,40 @@ export class GithubTarget extends BaseTarget {
           Accept: DEFAULT_CONTENT_TYPE,
         },
       });
+    } catch (e) {
+      throw new Error(
+        `Cannot get asset on GitHub. Status: ${(e as any).status}\n` + e
+      );
+    }
 
+    const etag = response.headers['etag'];
+    if (etag && etag.length > 0) {
       // ETag header comes in quotes for some reason so strip those
-      const remoteChecksum = response.headers['etag']?.slice(1, -1);
-      if (remoteChecksum) {
-        // XXX: This local checksum calculation only holds for simple cases:
-        // https://github.com/getsentry/craft/issues/322#issuecomment-964303174
-        const localChecksum = createHash('md5').update(file).digest('hex');
-        if (localChecksum !== remoteChecksum) {
-          logger.trace(`Checksum mismatch for "${name}"`, response.headers);
-          throw new Error(
-            `Uploaded asset MD5 checksum does not match local asset checksum for "${name} (${localChecksum} != ${remoteChecksum})`
-          );
-        }
-        uploadSpinner.succeed(`Uploaded asset "${name}".`);
-      } else {
-        logger.warn(`Response headers:`, response.headers);
-        uploadSpinner.succeed(
-          `Uploaded asset "${name}, but failed to find a remote checksum. Cannot verify uploaded file is as expected.`
-        );
-      }
-      return url;
-    } catch (e) {
-      uploadSpinner.fail(`Cannot upload asset "${name}".`);
+      return etag.slice(1, -1);
+    }
 
-      throw e;
+    return await this.md5FromUrl(url);
+  }
+
+  private async md5FromUrl(url: string): Promise<string> {
+    this.logger.debug('Downloading asset from GitHub to check MD5 hash: ', url);
+    let response;
+    try {
+      response = await this.github.request(`GET ${url}`, {
+        headers: {
+          Accept: DEFAULT_CONTENT_TYPE,
+        },
+      });
+    } catch (e) {
```

Comment:
I'm not sure if catching the original exception and throwing a new one is actually good here as you obscure the original issue. In debug mode, we already log the response status etc.

Reply:
Thanks @BYK !!

In this case here, we were seeing something like "Failed: Not Found" mixed up in the logs. So the `RequestError` (the `e`) from Octokit didn't provide much context, and not knowing clearly if the problem was in HEAD (the original ETag method) or GET (the new download method added in this PR) was not a great prospect.

The original error is still printed along with a custom message. Do you think we could do better?